File: java/api/src/main/java/org/dressdiscover/api/models/institution/InstitutionId.java
package org.dressdiscover.api.models.institution;

import org.thryft.waf.api.models.StringModelId;

import net.lab1318.costume.api.models.institution.InvalidInstitutionIdException;

public final class InstitutionId extends StringModelId {
    public static InstitutionId parse(final String value) throws InvalidInstitutionIdException {
        if (value.isEmpty()) {
            throw new InvalidInstitutionIdException("empty string");
        }
        return new InstitutionId(value);
    }

    private InstitutionId(final String value) {
        super(value);
    }
}

File: py/src/dressdiscover/api/models/institution/institution_id.py
from dressdiscover.api.models.institution.invalid_institution_id_exception import InvalidInstitutionIdException


class InstitutionId(str):
    @classmethod
    def parse(cls, str_):
        if len(str_) == 0:
            raise InvalidInstitutionIdException("empty")
        return cls(str_)

